test/Microsoft.ML.Tests/TrainerEstimators/TreeEstimators.cs

```diff
@@ -61,6 +61,124 @@ public void LightGBMBinaryEstimator()
             Done();
         }
 
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // LightGBM is 64-bit only
+        public void LightGBMBinaryEstimatorMonotoneThroughOptions()
```

Comment:
>LightGBMBinaryEstimatorMonotoneThroughOptions [](start = 20, length = 45)

Can we also have baseline tests?
we have tests like this: RegressorLightGBMTest
`var regPredictors = new[] { TestLearners.LightGBMReg };`
you can define new learner like
`  public static PredictorAndArgs LightGBMRegWithMonotone = new PredictorAndArgs
        {
            Trainer = new SubComponent("LightGBMR", "nt=1 iter=50 v=+ booster=gbdt{l1=0.2 l2=0.2} lr=0.2 mil=10 nl=20 MonotoneConstraints=pos:1-3 MonotoneConstraints=neg:4"),
            Tag = "LightGBMReg",
            BaselineProgress = true,
        };`
and add it to regPredictors.

Reply:
same for all types of trainers, but it shouldn't be hard I believe.

Reply:
added for Regression, MC and Classification

---
In reply to: [253163901](https://github.com/dotnet/machinelearning/pull/2330#discussion_r253163901) [](ancestors = 253163901)